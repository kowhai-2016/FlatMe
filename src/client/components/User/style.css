.User .user-row {
  margin-bottom: 14px;
}

.User .user-row:last-child {
  margin-bottom: 0;
}

.User .table-user-information > tbody > tr {
  border-top: 1px solid rgb(221, 221, 221);
}

.User .table-user-information > tbody > tr:first-child {
  border-top: 0;
}

.User .table-user-information > tbody > tr > td {
  border-top: 0;
}

.User .toppad {
  margin-top:20px;
}
