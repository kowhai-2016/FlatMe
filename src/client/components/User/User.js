import React, { PropTypes } from 'react'
import { Link } from 'react-router'

import './style.css'

const User = React.createClass({
  componentDidMount () {
    this.props.fetchUser()
  },
  render () {
    const user = this.props.user
    const title = user ? `${user.firstName} ${user.lastName}` : null
    const email = user ? `${user.email}` : null
    const phoneNumber = user ? `${user.phoneNumber}` : null
    return (
      <div className='User container'>
        <div className='row'>
          <div className='col-xs-12 col-sm-12 col-md-6 col-lg-6 col-xs-offset-0 col-sm-offset-0 col-md-offset-3 col-lg-offset-3 toppad' >
            <div className='panel panel-info'>
              <div className='panel-heading'>
                <h3 className='panel-title'>{title}</h3>
              </div>
              <div className='panel-body'>
                <div className='row'>
                  <div className='col-md-3 col-lg-3 '>
                    <img alt='User Pic' src='/images/profile.jpg' className='img-circle img-responsive' />
                  </div>
                  <div className=' col-md-9 col-lg-9 '>
                    <table className='table table-user-information'>
                      <tbody>
                        <tr>
                          <td>Email</td>
                          <td><a href={'mailto:' + `email`}>{email}</a></td>
                        </tr>
                        <tr>
                          <td>Phone Number:</td>
                          <td>{phoneNumber}</td>
                        </tr>
                        <tr>
                          <td>Home Address</td>
                          <td>Dev Academy</td>
                        </tr>
                      </tbody>
                    </table>
                    <Link to='/' className='btn btn-primary'>To Dashboard</Link>
                    <a href='#' className='btn btn-primary' style={{float: 'right'}}>Edit</a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
})

User.propTypes = {
  fetchUser: PropTypes.func.isRequired,
  user: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired
  })
}

export default User
