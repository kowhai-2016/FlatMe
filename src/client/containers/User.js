import { connect } from 'react-redux'

import { fetchUser } from '../actions'
import User from '../components/User'

const mapStateToProps = (state, ownProps) => {
  const id = Number(ownProps.params.id)
  return {
    user: state.users[id]
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    fetchUser: () => {
      const id = Number(ownProps.params.id)
      dispatch(fetchUser(id))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(User)
