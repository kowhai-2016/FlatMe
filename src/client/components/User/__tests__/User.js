/* eslint-env jest */

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

import User from '..'

test('User renders correctly', () => {
  const wrapper = shallow(<User />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
