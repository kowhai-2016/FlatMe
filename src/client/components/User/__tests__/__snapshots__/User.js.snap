exports[`test User renders correctly 1`] = `
<div>
  <h1>
    User
  </h1>
</div>
`;
